biorbd_optim/integrator.py

```diff
@@ -33,15 +34,15 @@ def dxdt(h, states, controls, params):
             x[:, i] = x[:, i - 1] + h / 6 * (k1 + 2 * k2 + 2 * k3 + k4)
 
         Quat_idx = []
-        for j in range(Model.nbQ()):
-            Name = Model.nameDof()[j].to_string()
+        for j in range(model.nbQ()):
+            Name = model.nameDof()[j].to_string()
             if Name[-5:-1] == "Quat":
                 Quat_idx += [j]
 
-        for j in range(Model.nbQuat()):
+        for j in range(model.nbQuat()):
```

Comment:
Sorry I did not pay enough attention. You can use
`biorbd_model.segment(idx).isRotationAQuaternion()`

So the first for loop probably can be reduce like so: 
```
quat_idx = [idx for idx in range(model.nbSegment()) if model.segment(idx).isRotationAQuaternion()]
```

Reply:
This would not give me the index of the quaternion Q, just the index of the quaternion joints.